source/image-handler/image-request.js

```diff
@@ -297,6 +298,48 @@ class ImageRequest {
 
         return null;
     }
+
+     /**
+      * Verifies base64 signature, using SIGNATURE_KEY env var. If SIGNATURE_KEY
+      * isn't provided, it doesn't do anything. Returns base64 path without the
+      * signature part.
+      * @param {String} path - base64-encoded URL path.
+      */
+     verifySignature(path) {
+        const signKey = process.env.SIGNATURE_KEY;
+        if (signKey === undefined) {
+          return path;
+        }
+ 
+        const crypto = require('crypto');
+        const bufferEq = require('buffer-equal-constant-time');
+ 
+        const splitPath = path.split("--");
+        const base64 = splitPath[0];
+        const providedSignature = splitPath[1];
+ 
+        if (providedSignature === undefined) {
+          throw ({
+              status: 400,
+              code: 'DecodeRequest::MissingSignature',
+              message: 'The signature is missing.'
+          });
+        }
+ 
+        const signature = crypto.createHmac("sha1", signKey).update(base64).digest("hex");
```

Comment:
SHA1 is broken. I know this is signing, not encryption, but I question this algorithm choice. I'd expect SHA-256 at minimum. What algorithm does AWS use for its S3 URL signing? I doubt it's SHA1.